src/View/Helper/UrlHelper.php

```diff
@@ -25,6 +27,43 @@
  */
 class UrlHelper extends Helper
 {
+    /**
+     * Default config for this class
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'assetUrlClassName' => Asset::class,
+    ];
+
+    /**
+     * Asset URL engine class name
+     *
+     * @var string
+     * @psalm-var class-string<\Cake\Routing\Asset>
+     */
+    protected $_assetUrlClassName;
+
+    /**
+     * Check proper configuration
+     *
+     * @param array $config The configuration settings provided to this helper.
+     * @return void
+     */
+    public function initialize(array $config): void
+    {
+        parent::initialize($config);
+        $engineClassConfig = $this->getConfig('assetUrlClassName');
+
+        /** @psalm-var class-string<\Cake\Routing\Asset>|null $engineClass */
+        $engineClass = App::className($engineClassConfig, 'Routing');
+        if ($engineClass === null) {
+            throw new CakeException(sprintf('Class for %s could not be found', $engineClassConfig));
+        }
+
+        $this->_assetUrlClassName = $engineClass;
```

Comment:
One problem I can see with having this being a config option and a separate property, is that after the helper is built someone could do `setConfig('assetUrlClassName', 'Something')` and it would take no effect. I don't think that will come up often, but it will be a 'bug' when it does.

Reply:
Yes, that can be an issue. The same issues exits for `TimeHelper` and `NumberHelper` too. We would have to override `setConfig()` to check the config keys and update the property too if require if we really want to address it.

Reply:
Well those haven't been a problem this far, so let's go with this.